### ISSUE TITLE
Test failures

### ISSUE THREAD

**Issue by ramonza** [census-ecosystem/opencensus-go-exporter-graphite#5]
See: https://api.travis-ci.org/v3/job/414752965/log.txt

```
Listening on 127.0.0.1:2003
==================
WARNING: DATA RACE
Write at 0x0000009e9310 by goroutine 16:
  contrib.go.opencensus.io/exporter/graphite.handleRequest()
      /home/travis/gopath/src/contrib.go.opencensus.io/exporter/graphite/graphite_test.go:99 +0x3fc

Previous write at 0x0000009e9310 by goroutine 12:
  contrib.go.opencensus.io/exporter/graphite.handleRequest()
      /home/travis/gopath/src/contrib.go.opencensus.io/exporter/graphite/graphite_test.go:99 +0x3fc

Goroutine 16 (running) created at:
  contrib.go.opencensus.io/exporter/graphite.startServer()
      /home/travis/gopath/src/contrib.go.opencensus.io/exporter/graphite/graphite_test.go:78 +0x2ef

Goroutine 12 (finished) created at:
  contrib.go.opencensus.io/exporter/graphite.startServer()
      /home/travis/gopath/src/contrib.go.opencensus.io/exporter/graphite/graphite_test.go:78 +0x2ef
==================
==================
WARNING: DATA RACE
Write at 0x000000a07bc5 by goroutine 8:
  contrib.go.opencensus.io/exporter/graphite.TestMetricsEndpointOutput()
      /home/travis/gopath/src/contrib.go.opencensus.io/exporter/graphite/graphite_test.go:156 +0xecf
  testing.tRunner()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/testing/testing.go:777 +0x16d

Previous read at 0x000000a07bc5 by goroutine 9:
  contrib.go.opencensus.io/exporter/graphite.startServer()
      /home/travis/gopath/src/contrib.go.opencensus.io/exporter/graphite/graphite_test.go:67 +0x2ff

Goroutine 8 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/testing/testing.go:777 +0x16d
  testing.runTests()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/testing/testing.go:978 +0x2cd
  main.main()
      _testmain.go:56 +0x22a

Goroutine 9 (running) created at:
  contrib.go.opencensus.io/exporter/graphite.TestMetricsEndpointOutput()
      /home/travis/gopath/src/contrib.go.opencensus.io/exporter/graphite/graphite_test.go:115 +0x319
  testing.tRunner()
      /home/travis/.gimme/versions/go1.10.3.linux.amd64/src/testing/testing.go:777 +0x16d
==================
--- FAIL: TestMetricsEndpointOutput (0.04s)
	testing.go:730: race detected during execution of test
=== RUN   TestMetricsTagsOutput
Error listening: listen tcp 127.0.0.1:2003: bind: address already in use
--- PASS: TestMetricsTagsOutput (0.05s)
2018/08/11 01:03:34 Failed to export to Graphite: Error creating graphite: &net.OpError{Op:"dial", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc420199200), Err:(*os.SyscallError)(0xc4200eca00)}
2018/08/11 01:03:34 Failed to export to Graphite: Error creating graphite: &net.OpError{Op:"dial", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc4201ccde0), Err:(*os.SyscallError)(0xc4200eca20)}
=== RUN   TestMetricsPathOutput
Listening on 127.0.0.1:2003
--- PASS: TestMetricsPathOutput (0.01s)
=== RUN   TestMetricsSumDataPathOutput
Error listening: listen tcp 127.0.0.1:2003: bind: address already in use
--- PASS: TestMetricsSumDataPathOutput (0.01s)
=== RUN   TestMetricsLastValueDataPathOutput
Error listening: listen tcp 127.0.0.1:2003: bind: address already in use
--- FAIL: TestMetricsLastValueDataPathOutput (0.02s)
	graphite_test.go:396: path not correct
=== RUN   TestDistributionData
Error listening: listen tcp 127.0.0.1:2003: bind: address already in use
2018/08/11 01:03:34 Failed to export to Graphite: Error creating graphite: &net.OpError{Op:"dial", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc4202c0bd0), Err:(*os.SyscallError)(0xc4200edaa0)}
2018/08/11 01:03:34 Failed to export to Graphite: Error creating graphite: &net.OpError{Op:"dial", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc4202c0b10), Err:(*os.SyscallError)(0xc4200edae0)}
2018/08/11 01:03:34 Failed to export to Graphite: Error creating graphite: &net.OpError{Op:"dial", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc4202c0ab0), Err:(*os.SyscallError)(0xc4200edb00)}
```

**Comment by ramonza** [comment]
@vcasadei could you take a look at the test failures on Travis? I see these locally as well when running:

`go test -race ./...`

from the root of this repo.

**Comment by ramonza** [comment]
For now I have ignored the failing tests: 
https://github.com/census-ecosystem/opencensus-go-exporter-graphite/pull/6

**Comment by vcasadei** [comment]
@ramonza working on it

**Comment by ramonza** [comment]
Resolved by #8
